cogs/poetry.py

```diff
@@ -0,0 +1,85 @@
+from os import path
+from pickle import load
+from typing import List, Tuple
+from random import choice
+from discord.ext import commands
+from .utils.poetry_toolz import PoetryGen, parse_poem_config
+
+
+class Poems(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        rel_path = path.abspath(path.dirname(__file__))
+        markov_file = open(path.join(rel_path, "utils/rev_markov"), "rb")
+        rhyme_file = open(path.join(rel_path, "utils/rhyme_dict"), "rb")
+        self.poem_machine = PoetryGen(load(markov_file), load(rhyme_file))
+        markov_file.close()
+        rhyme_file.close()
+
+    @commands.command()
+    async def poem(self, ctx, *, command: str = None):
+        """
+		Generate a poem from one of our prebuilt configurations using our (not) state of the poetry generator!
+		Either call "?poem" for a random configration, or call "?poem x" where x is one of the valid prebuilts.
+		Valid prebuilts are: haiku, limerick and alexandrine
+        """
+        configs: List[List[Tuple[str,
```

Comment:
better served by an immutable data structure, and if you use a trailing comma on the last item it can be aligned a bit nicer like
```
(
   [(), (), ()],
   [(), ()],
   ...
)
```

Reply:
in fact, this appears to really be a dictionary, since it corresponds to keys used below

Reply:
yep, agreed, made it into a dict